onnx/defs/traditionalml/defs.cc

```diff
@@ -915,131 +946,195 @@ ONNX_ML_OPERATOR_SET_SCHEMA(
+        .Input(0, "X", "Input of shape [Batch Size, Number of Features]", "T")
+        .Output(0, "Y", "Output of shape [Batch Size, Number of targets]", "T")
+        .TypeConstraint("T", {"tensor(float)", "tensor(double)"}, "The input type must be a tensor of a numeric type.")
+        .Attr("nodes_featureids", "Feature id for each node.", AttributeProto::INTS, true)
         .Attr(
-            "nodes_nodeids",
-            "Node id for each node. Node ids must restart at zero for each tree and increase sequentially.",
-            AttributeProto::INTS,
-            OPTIONAL_VALUE)
-        .Attr("nodes_featureids", "Feature id for each node.", AttributeProto::INTS, OPTIONAL_VALUE)
-        .Attr(
-            "nodes_values",
-            "Thresholds to do the splitting on for each node.",
-            AttributeProto::FLOATS,
-            OPTIONAL_VALUE)
-        .Attr(
-            "nodes_values_as_tensor",
-            "Thresholds to do the splitting on for each node.",
+            "nodes_splits",
+            "Thresholds to do the splitting on for each node with mode that is not 'SET_MEMBER'.",
             AttributeProto::TENSOR,
-            OPTIONAL_VALUE)
+            true)
         .Attr(
             "nodes_hitrates",
             "Popularity of each node, used for performance and may be omitted.",
-            AttributeProto::FLOATS,
-            OPTIONAL_VALUE)
-        .Attr(
-            "nodes_hitrates_as_tensor",
-            "Popularity of each node, used for performance and may be omitted.",
             AttributeProto::TENSOR,
             OPTIONAL_VALUE)
         .Attr(
             "nodes_modes",
-            "The node kind, that is, the comparison to make at the node. There is no comparison to make at a leaf node.<br>One of 'BRANCH_LEQ', 'BRANCH_LT', 'BRANCH_GTE', 'BRANCH_GT', 'BRANCH_EQ', 'BRANCH_NEQ', 'LEAF'",
-            AttributeProto::STRINGS,
-            OPTIONAL_VALUE)
-        .Attr("nodes_truenodeids", "Child node if expression is true", AttributeProto::INTS, OPTIONAL_VALUE)
-        .Attr("nodes_falsenodeids", "Child node if expression is false", AttributeProto::INTS, OPTIONAL_VALUE)
+            "The comparison operation performed by the node. This is encoded as an enumeration of 0 ('BRANCH_LEQ'), 1 "
+            "('BRANCH_LT'), 2 ('BRANCH_GTE'), 3 ('BRANCH_GT'), 4 ('BRANCH_EQ'), 5 ('BRANCH_NEQ'), and 6 "
+            "('SET_MEMBER').",
+            AttributeProto::INTS,
+            true)
         .Attr(
-            "nodes_missing_value_tracks_true",
-            "For each node, define what to do in the presence of a NaN: use the 'true' (if the attribute value is 1) or 'false' (if the attribute value is 0) branch based on the value in this array.<br>This attribute may be left undefined and the default value is false (0) for all nodes.",
+            "nodes_truenodeids",
+            "If `nodes_trueleafs` is false at an entry, this represents the position of the true branch node. This "
```

Comment:
What would you do if the truenode is a leaf and the false node is a node?

Reply:
These would be the entries if that was the case for the nth node:
nodes_truenodeids[i] = position of leaf in leaf_* attributes
nodes_trueleafs[i] = 1
nodes_falsenodeids[i] = position of node in nodes_* attributes
nodes_falseleafs[i] = 0

Reply:
In the example [here](https://github.com/onnx/onnx/pull/5874/files#diff-688311d1988fd5cf6c54f96217dcf7ff8eb9600c771fb83c1ad57e7b48309572R85-R88), the first node has a leaf in the false branch and an interior node in the right branch.